ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -335,32 +335,46 @@ def _router_bgp_vlans(self) -> list | None:
         vlans = []
         for tenant in self.shared_utils.filtered_tenants:
             for vrf in tenant["vrfs"]:
-                for svi in vrf["svis"]:
-                    if (vlan := self._router_bgp_vlans_vlan(svi, tenant, vrf)) is not None:
-                        vlan_id = int(svi["id"])
-                        vlan = {"id": vlan_id, **vlan}
-                        append_if_not_duplicate(
-                            list_of_dicts=vlans,
-                            primary_key="id",
-                            new_dict=vlan,
-                            context="BGP VLANs defined under network services",
-                            context_keys=["id", "tenant"],
-                            ignore_keys={"tenant"},
-                        )
+                sorted_vlan_list = sorted(vrf["svis"], key=self._get_vlan_aware_bundle_name_tuple_for_svis)
+                bundle_groups = itertools_groupby(sorted_vlan_list, self._get_vlan_aware_bundle_name_tuple_for_svis)
+
+                for vlan_aware_bundle_name_tuple, svis in bundle_groups:
+                    bundle_name, is_evpn_vlan_bundle = vlan_aware_bundle_name_tuple
+                    svis = list(svis)
+                    if not is_evpn_vlan_bundle:
```

Comment:
To avoid doing these same operations multiple times, we may have to build some common function to split the vlans in two. Can I ask you to perform a performance comparison with/without this PR on the largest inventory you have.

Reply:
updated code